src/trace/entry.zig

```diff
@@ -1,245 +1,87 @@
-                std.debug.print("Processing fields failed with err: {any}", .{err});
-                self.deinit();
-                return;
-            },
-            .maybe_fmt = null,
-        };
-
-        self.channel.send(log_msg) catch |err| {
-            std.debug.print("Send msg through channel failed with err: {any}", .{err});
-            self.deinit();
-            return;
-        };
-    }
-
-    pub fn logf(self: *Self, comptime fmt: []const u8, args: anytype) void {
-        // Get memory for formatting message.
-        const msg_buf = self.allocBuf(std.fmt.count(fmt, args)) catch |err| {
-            std.debug.print("allocBuff failed with err: {any}", .{err});
-            self.deinit();
-            return;
-        };
-        var fmt_message = std.io.fixedBufferStream(msg_buf);
-        // Format message.
-        logfmt.fmtMsg(fmt_message.writer(), fmt, args);
-
-        const log_msg = logfmt.LogMsg{
-            .level = self.log_level,
-            .maybe_scope = self.scope,
-            .maybe_msg = null,
-            .maybe_fields = self.fields.toOwnedSlice() catch |err| {
-                std.debug.print("Processing fields failed with err: {any}", .{err});
-                self.deinit();
-                return;
-            },
-            .maybe_fmt = fmt_message.getWritten(),
-        };
-
-        self.channel.send(log_msg) catch |err| {
-            std.debug.print("Send msg through channel failed with err: {any}", .{err});
-            self.deinit();
-            return;
-        };
-    }
-
-    // Utility function for allocating memory from RecycleFBA for part of the log message.
-    fn allocBuf(self: *const Self, size: u64) ![]u8 {
-        const buf = blk: while (true) {
-            const buf = self.allocator.alloc(u8, size) catch {
-                std.time.sleep(std.time.ns_per_ms);
-                if (self.exit_sig.load(.unordered)) {
-                    return error.MemoryBlockedWithExitSignaled;
-                }
-                continue;
+        /// Returns a new struct type based on Fields, just with one more field added.
+        fn FieldsPlus(comptime field_name: [:0]const u8, comptime FieldType: type) type {
+            const info = @typeInfo(Fields);
+            var new_fields: [1 + info.Struct.fields.len]std.builtin.Type.StructField = undefined;
+            for (info.Struct.fields, 0..) |existing_field, i| {
+                new_fields[i] = existing_field;
+            }
+            const ActualFieldType = switch (@typeInfo(FieldType)) {
+                .ComptimeFloat => f64,
+                .ComptimeInt => u64,
+                else => FieldType,
+            };
+            new_fields[info.Struct.fields.len] = .{
+                .name = field_name,
+                .type = ActualFieldType,
+                .default_value = null,
+                .is_comptime = false,
+                .alignment = @alignOf(FieldType),
             };
```

Comment:
I know this is in draft, but just want to quickly provide this suggestion:
```suggestion
            const ActualFieldType = switch (@typeInfo(FieldType)) {
                .ComptimeFloat => f64,
                .ComptimeInt => u64,
                else => FieldType,
            };
            const new_fields = info.Struct.fields ++ &[_]std.builtin.Type.StructField{.{
                .name = field_name,
                .type = ActualFieldType,
                .default_value = null,
                .is_comptime = false,
                .alignment = 0,
            }};
```
And then also change the line below from `.fields = &new_fields` to `.fields = new_fields` - github is stupid and won't let me include the lines below this one in the suggestion.
```

Reply:
80e603c7cc88d00e324a4ed85602df2afcad2cfe